src/OrderHandler.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {ComposableCoW} from "cow-order/ComposableCoW.sol";
+import {BaseConditionalOrder} from "cow-order/BaseConditionalOrder.sol";
+
+import {GPv2Order} from "cowprotocol/libraries/GPv2Order.sol";
+
+import {IERC20} from "@openzeppelin/token/ERC20/IERC20.sol";
+
+import {IChronicleOracle} from "./interfaces/IChronicleOracle.sol";
+
+/// @title OrderHandler
+/// @notice Handler allows to ensure that the order details are executed according to onchain conditions.
+/// @dev Designed to be used with the CoW Protocol Conditional Order Framework.
+contract OrderHandler is BaseConditionalOrder {
+    /*//////////////////////////////////////////////////////////////////////////
+                                   CONSTANTS
+    //////////////////////////////////////////////////////////////////////////*/
+
+    // ref: https://chroniclelabs.org/dashboard/oracle/ETH/USD?blockchain=GNO&txn=0x553acf4265afea825633c10a3252e7ab532c655745a11c47d4cdff8d3714b725&contract=0x5e16ca75000fb2b9d7b1184fa24ff5d938a345ef
+    IChronicleOracle constant ORACLE_CHRONICLE = IChronicleOracle(0x5E16CA75000fb2B9d7B1184Fa24fF5D938a345Ef);
+
+    ComposableCoW public immutable composableCow;
+
+    constructor(ComposableCoW _composableCow) {
+        composableCow = _composableCow;
+    }
+
+    function kissMe() external {
```

Comment:
is this allowed on gnosis mainnet?

Reply:
afaik you can do the WL in 2 ways one is thru the self-kiss SC which basically calls the `kiss` method of the oracle itself as a helper or directly go into the oracle with an auth addy (which i guess is their devs) and get auth by impersonating them in the meantime

Reply:
resolved in 91def8a15a3aa4fa451cf2dea4b5ed606e8f7959